executor/runtime/docker/docker_linux.go

```diff
@@ -286,29 +336,14 @@ func setCgroupOwnership(parentCtx context.Context, c runtimeTypes.Container, cre
 	}
 
 	cgroupPath := filepath.Join("/proc/", strconv.Itoa(int(cred.pid)), "cgroup")
-	cgroups, err := ioutil.ReadFile(cgroupPath) // nolint: gosec
+	cgroupsInfo, err := cgroups.ParseCgroupFile(cgroupPath)
 	if err != nil {
 		logrus.WithError(err).Error("Could not read container cgroups")
 		return err
 	}
-	for _, line := range strings.Split(string(cgroups), "\n") {
-		cgroupInfo := strings.Split(strings.TrimSpace(line), ":")
-		if len(cgroupInfo) != 3 {
-			continue
-		}
-		controllerType := cgroupInfo[1]
-		if len(controllerType) == 0 {
-			continue
-		}
-		// This is to handle the name=systemd cgroup, we should probably parse /proc/mounts, but this is a little bit easier
-		controllerType = strings.TrimPrefix(controllerType, "name=")
-		if controllerType != "systemd" {
-			continue
-		}
+	if controllerPath, ok := cgroupsInfo["name=systemd"]; ok {
 
-		// systemd needs to be the owner of its systemd cgroup in order to start up
-		controllerPath := cgroupInfo[2]
-		fsPath := filepath.Join(sysFsCgroup, controllerType, controllerPath)
+		fsPath := filepath.Join(sysFsCgroup, "systemd", controllerPath)
```

Comment:
Are you sure the cleanupCgroups function doesn't need to change at all due to this change?

Reply:
systemd automatically cleans up the cgroup hierarchy once the daemon dies.

Reply:
Make sure the executor unit has Delegate=True.

Reply:
Set to True in the current config. https://github.com/Netflix/titus-executor/blob/master/root/lib/systemd/system/titus-executor%40.service#L16